### DISCUSSION TITLE
Prompt choices dynamically

### DISCUSSION THREAD

**Discussion by joserocha** [spectreconsole/spectre.console#167] [Q&A] [answered]
Is there some way to add choices dynamically ?
I tried every possibility I know (which is not much) but no success.

**Comment by patriksvensson** [comment, marked as answer]
@joserocha Not sure I understand what you mean, but does something like this solve it for you?

```csharp
// Build the fruit list
var fruits = new List<string>();
fruits.Add("banana");
fruits.Add("orange");

// Build the prompt
var prompt = new TextPrompt<string>("What's your [green]favorite fruit[/]?");
foreach(var fruit in fruits) {
   prompt.AddChoice(fruit);
}

// Show the prompt
var selectedFruit = AnsiConsole.Prompt(prompt);
```

**Reply by joserocha** [reply]
That's exactly what I need =) ... Thank you !!

**Comment by Sachin-Nand** [comment]
Can AddChoices option of SelectionPrompt made dynamic ?

**Reply by Sachin-Nand** [reply]
Ok I have figured it out

        var prompt = new SelectionPrompt<string>();
        foreach (var lkhouse in lakehouse)
        {
        prompt.AddChoice(lakehouse[i].ToString());
        i++;
        }
        prompt.Title("Which lakehouse[green]would you like to interact with[/]?")
        .PageSize(10)
        .MoreChoicesText("[grey](Move up and down to reveal more lakehouses)[/]");
        
        var lakehouse = AnsiConsole.Prompt(prompt);
